cleaner.rb

```diff
@@ -1,34 +1,49 @@
 require 'aws-sdk'
 require 'awesome_print'
+require 'time'
 
 akid = ENV.fetch "AWS_ACCESS_KEY_ID"
 secret = ENV.fetch "AWS_SECRET_ACCESS_KEY"
 region = ENV.fetch "AWS_DEFAULT_REGION"
+age = ENV.fetch "AGE_DAYS"
+
+clean_time = Time.now - age.to_i * 86400
 
 Aws.config.update({
   region: region,
   credentials: Aws::Credentials.new(akid, secret),
 })
 
-beanstalk = Aws::ElasticBeanstalk::Client.new
+client = Aws::ElasticBeanstalk::Client.new
 
-resp = beanstalk.describe_applications
+resp = client.describe_applications
 resp.applications.each do |app|
   app.versions.each do |version|
-    env = beanstalk.describe_environments({
+    version_info = client.describe_application_versions({
```

Comment:
Instead of describing a version at a time, what do you think about describing all the versions first and then doing a `.each` on those. Perhaps something like (pardon my ruby, it's been a while)

```ruby
resp.applications.each do |app|
  options = { application_name: app.application_name, version_labels: app.versions }
  resp = client.describe_application_versions(options)
  versions = resp.application_versions;
  while resp. next_token do
    options. next_token = resp.next_token;
    resp = client.describe_application_versions(options)
  end

  versions.each do |version|
    # check the date and delete if it is old enough
  end 
end
```